api/proposal/by-document.go

```diff
@@ -12,54 +12,57 @@ import (
 	"golang.org/x/exp/slog"
 )
 
-func ProposalByDocumentHandler(res http.ResponseWriter, req *http.Request) {
-	switch req.Method {
-	case http.MethodDelete:
-		deleteProposalByDocumentHandler(res, req)
-	case http.MethodGet:
-		getProposalByDocumentHandler(res, req)
-	default:
-		httputil.WriteError(res, http.StatusMethodNotAllowed, "invalid method")
-	}
-}
-
-func getProposalByDocumentHandler(res http.ResponseWriter, req *http.Request) {
-	// Check method GET is used
-	if req.Method != http.MethodGet {
-		httputil.WriteError(res, http.StatusMethodNotAllowed, "invalid method")
+// @Summary		Get proposals by document id
+// @Description	Get all proposals for a document, given its ID
+// @Tags			proposal
+// @Param			id	path	string	true	"Document id"
+// @Produce		json
+// @Success		200	{object}	DocumentProposal
+// @Failure		400	{object}	httputil.ApiError
+// @Router			/proposals/document/{id} [get]
+func GetProposalByDocumentHandler(res http.ResponseWriter, req *http.Request) {
+	if !middleware.GetAdmin(req) {
+		httputil.WriteError(res, http.StatusForbidden, "you are not admin")
 		return
 	}
+
 	db := util.GetDb()
 	docID := muxie.GetParam(res, "id")
+
 	var questions []Proposal
 	if err := db.Where(models.Question{Document: docID}).Find(&questions).Error; err != nil {
```

Comment:
I'd like to stay consistent as much as possible. Here you're using:
```
	db.Where(models.Question{Document: docID})
```
But later you decided to use:
```
	db.Where("document = ?", docID})
```
Pick one way (for me both are equivalent) but stick to only of of them :)